File: content-api/orchestrator/src/main/java/org/sunbird/actors/CollectionActor.java
package org.sunbird.actors;

import org.sunbird.actor.core.BaseActor;
import org.sunbird.common.dto.Request;
import org.sunbird.common.dto.Response;
import org.sunbird.managers.HierarchyManager;
import org.sunbird.managers.UpdateHierarchyManager;
import scala.concurrent.Future;

public class CollectionActor extends BaseActor {

    private static final String SCHEMA_NAME = "collection";
    
    @Override
    public Future<Response> onReceive(Request request) throws Throwable {
        String operation = request.getOperation();
        switch (operation) {
            case "addHierarchy": return addLeafNodesToHierarchy(request);
            case "removeHierarchy": return removeLeafNodesFromHierarchy(request);
            case "updateHierarchy": return updateHierarchy(request);
            case "getHierarchy": return getHierarchy(request);
            default: return ERROR(operation);
        }
    }

    private Future<Response> addLeafNodesToHierarchy(Request request) throws Exception {
        request.getContext().put("schemaName", SCHEMA_NAME);
        return HierarchyManager.addLeafNodesToHierarchy(request, getContext().dispatcher());
    }

    private Future<Response> removeLeafNodesFromHierarchy(Request request) throws Exception {
        request.getContext().put("schemaName", SCHEMA_NAME);
        return HierarchyManager.removeLeafNodesFromHierarchy(request, getContext().dispatcher());
    }

    private Future<Response> updateHierarchy(Request request) throws Exception {
        request.getContext().put("schemaName", SCHEMA_NAME);
        return UpdateHierarchyManager.updateHierarchy(request, getContext().dispatcher());
    }
    private Future<Response> getHierarchy(Request request) throws Exception {
        request.getContext().put("schemaName", SCHEMA_NAME);
        return HierarchyManager.getHierarchy(request, getContext().dispatcher());
    }
}

File: content-api/content-actors/src/main/scala/org/sunbird/content/actors/CollectionActor.scala
package org.sunbird.content.actors

import javax.inject.Inject
import org.sunbird.actor.core.BaseActor
import org.sunbird.common.dto.{Request, Response}
import org.sunbird.graph.OntologyEngineContext
import org.sunbird.managers.{HierarchyManager, UpdateHierarchyManager}
import org.sunbird.utils.HierarchyConstants

import scala.concurrent.{ExecutionContext, Future}

class CollectionActor @Inject() (implicit oec: OntologyEngineContext) extends BaseActor {

    implicit val ec: ExecutionContext = getContext().dispatcher

    override def onReceive(request: Request): Future[Response] = {
        request.getContext.put(HierarchyConstants.SCHEMA_NAME, HierarchyConstants.COLLECTION_SCHEMA_NAME)
        request.getOperation match {
            case "addHierarchy" => HierarchyManager.addLeafNodesToHierarchy(request)
            case "removeHierarchy" => HierarchyManager.removeLeafNodesFromHierarchy(request)
            case "updateHierarchy" => UpdateHierarchyManager.updateHierarchy(request)
            case "getHierarchy" => HierarchyManager.getHierarchy(request)
            case _ => ERROR(request.getOperation)
        }
    }

}
